### ISSUE TITLE
Installing VSCode v1.86 broke debug breakpoints

### ISSUE THREAD

**Issue by brucejo75** [microsoft/vscode#204784]
Does this issue occur when all extensions are disabled?: Yes/No
 **Yes** still occurs with all extension disabled

- VS Code Version: 1.86.1
- OS Version: Wndows 10 22H2 (OS Build 19045.3996)

## Steps to Reproduce:

1. Install v1.86.1
2. Run `node` process, connect VSCode as debugger, try to set a breakpoint.
3. Note that breakpoints have empty circle not red dot.

## Notes
1. There is a new panel, `loaded scripts`, that is empty.

2. Node debugging with Chrome works fine.
   I set a breakpoint in Chrome debugger separately.
   Noticed that my connected session in VSCode also hit the breakpoint and highlighted the correct line of code.

I think the debugger simply lost the path to my code for setting breakpoints.

<details>

<summary>My "launch.json" config</summary>

```json
    {
      "name": "Meris.meris",
      "port": 5838,
      "request": "attach",
      "skipFiles": [
        "<node_internals>/**"
      ],
      "type": "node",
      "sourceMaps": true,
      "remoteRoot": "${workspaceFolder}/meris",
      "localRoot": "${workspaceFolder}/meris",
      "sourceMapPathOverrides": {
        "meteor://💻app/packages/*": "${workspaceFolder}/packages/*",  // Example: "meteor://💻app/main.ts" -> "c:/code/main.ts"
        "meteor://💻app/*": "${workspaceFolder}/meris/*"  // Example: "meteor://💻app/main.ts" -> "c:/code/main.ts"
      },
    },
```
</details>

## Workaround
go back to v1.85. All works fine.

Had to uninstall GitHub CoPilot to get CoPilot working again. (yeah, I know another issue)

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! It looks like you may be using an old version of VS Code, the latest stable release is 1.86.1. Please try upgrading to the latest version and checking whether this issue remains.

Happy Coding!

**Comment by brucejo75** [comment]
OK, went back and verified that v1.86.1 still fails. And I updated all the comments in the initial issue comment.

**Comment by sebestindragos** [comment]
Up! Also facing this issue, my entire day went down the drain trying to figure out why they are not working

**Comment by sebestindragos** [comment]
@brucejo75 downgrading to 1.85.2 until this is fixed, works on my end. You can get it from here https://code.visualstudio.com/updates/v1_85

**Comment by VSCodeTriageBot** [comment]
Please collect trace logs using the following instructions:

> If you're able to, add `"trace": true` to your `launch.json` and reproduce the issue. The location of the log file on your disk will be written to the Debug Console. Share that with us.
>
> ⚠️ This log file will not contain source code, but will contain file paths. You can drop it into https://microsoft.github.io/vscode-pwa-analyzer/index.html to see what it contains. If you'd rather not share the log publicly, you can email it to connor@xbox.com

**Comment by DavidKirn** [comment]
The breakpoints work for me, but sometimes the debug control panel 
![grafik](https://github.com/microsoft/vscode/assets/65389717/972da47a-1cf8-42a5-bb5d-8f6613ccd78d)
disappears during debugging even though the debug session is still running. Only after a vs code restart everything works normally again. But then the error occurs again after an undefined time or action.

**Comment by brucejo75** [comment]
Here is the log.

What I did:
1. connected to my node server.
2. set a breakpoint at a location I know I was going to hit (the breakpoint showed up as an unfilled grey circle, not red circle)
3. Ran my code

Expected: breakpoint to be hit
Actual: no breakpoint hit

Log file: [vscode-debugadapter-d7a9e62e.json.gz](https://github.com/microsoft/vscode/files/14335525/vscode-debugadapter-d7a9e62e.json.gz)

**Comment by connor4312** [comment]
In the log you provided, you set breakpoints in files named handler.js, server.js, and upload.js, however I'm not seeing any files with these names get loaded by your program -- the breakpoints show as unverified unless the associated source gets loaded.

Are you able to share a log from 1.85 as well so I can see how things 'should' be working?

**Comment by brucejo75** [comment]
Hi @connor4312,

Ran the same test on v`1.85.2`. This time the breakpoint in `handlers.js` was hit.

Log file: [vscode-debugadapter-ac6fd48a.json.gz](https://github.com/microsoft/vscode/files/14349079/vscode-debugadapter-ac6fd48a.json.gz)

**Comment by connor4312** [comment]
Thanks. I'm actually not sure why it was working before; it shouldn't have been. You should be able to resolve this by setting `"resolveSourceMapLocations": null` in your launch.json. 

It seems like meteor sets the URL of compiled files, not just source files (e.g. `meteor://💻app/packages/meris.data-tables.js` is the URL in the scriptParsed event.) We could solve this in the debugger by applying sourceMapPathMappings to scripts with sourceURL's as well as sourcemaps. I think that should be quite safe.

**Comment by brucejo75** [comment]
[That ](https://github.com/microsoft/vscode/issues/204784#issuecomment-1954811816)does the trick!  Thanks!

# FIX
add `"resolveSourceMapLocations": null` to my `launch.json`.

**Comment by connor4312** [comment]
I'll keep this issue open to track the fix I mentioned that we could do on the debugger side
